machine-prefs: insensitive read-only property

dlbeswick opened the pull request:
Hello, two small commits to do with machine prefs here.

The first is that the wrong kind of data type was being passed to gtk_list_store_new, causing a stack overwrite.

The second is just a minor tweak to make sure that widgets showing read-only machine properties are made "insensitive" (disabled.)

ensonic commented:
Should we maybe rather skip the non-writable ones (and not show them at all?) Do you have an example? Another option might be to just show them as a label so that the user won't wonder why it is not editable?

Thanks for the other fix with the type mixup. If you move this int a separate PR, I can merge it right away.

dlbeswick commented:
Here's what I see with pulsesrc:

![image](https://user-images.githubusercontent.com/907688/110224348-da24cc80-7f2e-11eb-82fe-7689deebab44.png)

When you say "show them as label", do you mean just show the property name and not the value? I think it's still useful to show the values, particularly to help debug problems. "Preferences" like these tend to be the advanced or lesser-used properties of the machines anyway, I feel. But let me know what you'd prefer; I'm not fussy and can change it.

Another option could be to prepend the tooltip text with a string like "(Read-only) ".

ensonic commented:
Thanks for the screenshot. I mean to also use a label for the value.
For the background, the preferences are setting on the element that cannot be changes over time as part of the score (in the pattern table).

dlbeswick commented:
Hello Stefan, I did have a quick crack at this, but personally I feel the logic was getting a little complicated for the payoff (or I don't know how to do it in a non-messy way.) I also felt in the end that using a label could make things more inconsistent -- for example, a writable boolean property would have a checkbox but a read-only boolean property would have a label with a value reading "false", I suppose.

You're right of course about the exact purpose of the preferences. This would only be an issue for the components that weren't specifically written for Buzztrax, I'd expect.

I've added a "(Read-only) " string to the tooltips and I found that works quite well, but apart from that I might just leave the change here at this point; happy for you to decide what to do with it.

I have split the other fix into a another pull request which I'll open shortly. Thanks!

ensonic commented:
You are right - checkboxes are a good example that might look weird to just see true/false (some might make more sense to people if called on/off). So lets go with this.
